polar-core/src/numerics.rs

```diff
@@ -46,7 +46,21 @@ where
         type Value = f64;
 
         fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-            formatter.write_str("JSON encoded data")
+            formatter.write_str("An integer (42), a float (1.2), or a string (\"NaN\")")
```

Comment:
We should probably be implementing a visitor for `Numeric` instead of just for float. I think that is closer to how JSON represents them - numbers are either float or ints.

Reply:
https://docs.serde.rs/serde_json/struct.Number.html is basically what we're working with